gem/lib/pupperware/spec_helper.rb

```diff
@@ -192,7 +192,15 @@ def wait_on_stack_healthy()
     # * all containers are healthy, none error and the wait is a success
     # * one container fails, and throws an exception, teling all the others to cancel
     #   and the exception is immediately raised here (other threads will gc)
-    ThreadsWait.all_waits(threads)
+    waiter = ThreadsWait.new(threads)
+    begin
+      waiter.all_waits()
+    rescue
+      # wait for all other threads to cancel so logs are ordered correctly
+      waiter.threads(&:join)
+      # re-raise the exception from first failing thread
+      raise
+    end
```

Comment:
Testing GitHub's new multiline comment.

....

Great work here 😄

Reply:
Updated to handle eating additional thread exceptions in the case where multiples fail simultaneously... should be good to merge after that goes through.